frontend/view-edit-client/interactions/service-interaction-inputs.component.tsx

```diff
@@ -0,0 +1,263 @@
+        dateOfInteraction,
+        duration: duration.stringValue,
+        location: selectedLocation,
+        description,
+      };
+
+      const apiPath = options.edit
+        ? `/api/clients/${props.clientId}/interactions/${props.initialInteraction.id}`
+        : `/api/clients/${props.clientId}/interactions`;
+      const apiUrl =
+        apiPath +
+        getTagsQuery(
+          userMode.userMode,
+          interaction.serviceId,
+          props.servicesResponse
+        );
+
+      return easyFetch(apiUrl, {
+        method: options.edit ? "PATCH" : "POST",
+        body: interaction,
+        signal,
+      });
+    },
+  }));
+
+  useEffect(() => {
+    props.setName(selectedService ? selectedService.serviceName : "");
+  }, [selectedService, props.setName]);
+
+  React.useEffect(() => {
+    if (selectedInteractionType === InteractionType.byPhone) {
+      setSelectedLocation("CUOffice");
+    }
+  }, [selectedInteractionType]);
+
+  React.useEffect(() => {
+    if (props.servicesResponse && !selectedService) {
+      const serviceId = props.initialInteraction
+        ? props.initialInteraction.serviceId
+        : null;
+
+      setSelectedService(
+        serviceId
+          ? props.servicesResponse.services.find((s) => s.id === serviceId)
+          : null
+      );
+    }
+  }, [props.servicesResponse, selectedService]);
+
+  React.useEffect(() => {
+    if (!selectedService) {
+      return;
+    } else {
+      if (selectedService.defaultInteractionLocation) {
+        setSelectedLocation(selectedService.defaultInteractionLocation);
+      }
+
+      if (selectedService.defaultInteractionType) {
+        setSelectedInteractionType(selectedService.defaultInteractionType);
+      }
+
+      if (selectedService.defaultInteractionDuration) {
+        setDuration({
+          stringValue: selectedService.defaultInteractionDuration,
+          hours: null,
+          minutes: null,
+        });
+      }
+    }
+  }, [selectedService]);
+
+  return (
+    <>
+      <label id={`provided-service-${props.interactionIndex}`}>Service:</label>
+      <div>
+        <select
+          value={selectedService ? selectedService.id : ""}
+          onChange={(evt) => {
+            const serviceId = Number(evt.target.value.slice("CU".length));
+            setSelectedService(services.find((s) => s.id === serviceId));
+          }}
+          aria-labelledby={`provided-service-${props.interactionIndex}`}
+          className="services-select"
+          name={`provided-service-${props.interactionIndex}`}
+          required
+        >
+          <option value="" disabled hidden>
+            Choose here
+          </option>
+          {Object.keys(groupedServices).map((programName) => (
+            <optgroup label={programName} key={programName}>
+              {groupedServices[programName].map((service) => (
+                <option key={service.id} value={"CU" + service.id}>
+                  {service.serviceName}
```

Comment:
State for selected service is correct but when option is chosen it quickly changes/defaults to displaying "Chronic Disease Screening" instead of the chosen service.

Reply:
Great catch - this was a bug. I fixed it in https://github.com/JustUtahCoders/comunidades-unidas-internal/pull/677/commits/07a28dcb099f92768122ca9405cea1b829e3fc17